cobudget/cobudget-api
# Prepare DB changes for moving to immutable transactions

Finally it's finished. It's been quite a lot of work and become a huge change. I know it's hard to review these really big changes, so I'm sorry about that.

Below is a few hints to make testing easier. 

# To test this

* Download the latest production database dump from the S3 bucket `database-dumps.cobudget.co` 
* If needed, change the Owner in the DB dump using sed:
	*  `sed -e 's/Owner: cobudget/Owner: <your-db owner>/' -e 's/OWNER TO cobudget/OWNER TO <your-db-owner>/' cobudget-prod-171122 >db_loadable`
* Reload the database with the newest production DB
	* `dropdb cobudget_development`
	* `createdb cobudget_development`
	* `psql -d cobudget_development <db_loadable`
* Migrate `rake db:migrate`
* Now you can check that all buckets and all memberships are correct after the migration (from the rails console):
	* `BucketService.check_all_buckets`
	* `MembershipService.check_all_memberships`
* Check any other things you feel is necessary

# Other comments

* Funds added to buckets that are funding or funded will no longer be removed when a member is archived, so a small change in the front end is needed as well.
* I'm happy to go through all of this in phone call if that's helpful - or help in any other way.
* Another way to do this, would be for you to point out whatever is unclear, which would give me an opportunity to make the code more readable and possibly add more comments @jlopker
